### ISSUE TITLE
All panel are getting collapsed when changing views

### ISSUE THREAD

**Issue by mtszsobczak** [darktable-org/darktable#3759]
**Describe the bug**
Every time I change a view eg. lighttable -> darkroom or inside lighttable I change option from file manager to culling all panels are getting collapsed so that I need to reopen them :(. This happened first for darkroom when I upgraded to one of dev versions. Yesterday when I closed panels and closed darktable and today after upgrading to latest dev version I have closed panels also on lighttable. Zoomable light table is working and its restoring all panels.

**To Reproduce**
1. Open darktable, all panels are closed on lighttable view
2. Reopen panels
2. Click on 'culling' mode and see that all panels are closed now
3. Reopen panels
4. Go to lighttable mode and see that all panels are closed now
4. Reopen panels
5. Go to darkroom and see that all panels are closed now
5. Go to lighttable
5. Reopen panels
5. Click 'W' in lighttable to Zoom image, after releasing the key see that all panels are collapsed

**Expected behavior**
Panels should be closed only in culling mode (I guess) and restored to last state when reentering other modes e.g. darkroom or file manager inside lighttable.

**Screenshots**
This is how I see darktable on every mode change (though with some images/image depending on mode. I filtered out images for this screenshot)
![image](https://user-images.githubusercontent.com/9355059/70860100-84c16180-1f1d-11ea-87cc-53bb3af79dd4.png)

**Version:**
![image](https://user-images.githubusercontent.com/9355059/70860156-4d9f8000-1f1e-11ea-9e9c-f3d31759a6ad.png)

**Platform (please complete the following information):**
 - OS: Ubuntu
 - Version: 19.10
 - OpenC: no

**Additional context**
 - Is dt compiled by yourself? If so which compiler is used? Which options?
```
 ~/projects/darktable   master ●  llvm-config --version
9.0.0
 ~/projects/darktable   master ●  gcc --version
gcc (Ubuntu 9.2.1-9ubuntu2) 9.2.1 20191008
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
```
diff --git a/cmake/compiler-warnings.cmake b/cmake/compiler-warnings.cmake
index c53f5371b..c79d66e1d 100644
--- a/cmake/compiler-warnings.cmake
+++ b/cmake/compiler-warnings.cmake
@@ -41,7 +41,8 @@ CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wno-format-truncation)
 
 # clang-4.0 bug https://llvm.org/bugs/show_bug.cgi?id=28115#c7
 CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wno-error=address-of-packed-member)
-
+CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wno-error=deprecated-declarations)
+CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wno-error=format-truncation)
 # should be < 64Kb
 math(EXPR MAX_MEANINGFUL_SIZE 32*1024)
 CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wframe-larger-than=${MAX_MEANINGFUL_SIZE})
(END)
```
* `./build.sh --prefix /opt/darktable --build-type Release`
* `sudo cmake --build "/home/mateusz/projects/darktable/build" --target install -- -j4`

Maybe I can clean some config which was badly generated during dev process and I am not aware of it?
Thanks for helping out! :cat:

**Comment by AlicVB** [comment]
hi, strange... and this happen suddenly ? Normally, panel state are stored per view/layout so that means that you can set different panel state for each view/layout and dt should remember them.
Can you share your darktablerc file ?

**Comment by mtszsobczak** [comment]
Sure, I am guessing that this is caused by following master and compiling frequently. It started to happen only for darkroom after first version of new styles was merged. And yesterday it expanded to light table. I remember that I closed all panels and then quit darktable. Today I compiled new version of darktable and now all panels are closed in darkroom/light table.

I am using i3 with mate if this have something to do with it.

my darktablerc file
```
bauhaus/scale=1.4
database=library.db
panel_width=350
slideshow_delay=5
ui_last/no_april1st=true
cache_memory=(1024 * 1024 * 512)
cache_disk_backend=true
cache_disk_backend_full=false
cache_color_managed=true
worker_threads=2
host_memory_limit=1500
singlebuffer_limit=16
opencl_memory_headroom=300
opencl_avoid_atomics=false
opencl_number_event_handles=25
opencl_async_pixelpipe=false
opencl_micro_nap=1000
opencl_device_priority=*/!0,*/*/*
opencl_mandatory_timeout=200
opencl_use_pinned_memory=false
opencl_use_cpu_devices=false
opencl_disable_drivers_blacklist=false
opencl_checksum=
pixelpipe_synchronization_timeout=200
never_use_embedded_thumb=false
write_sidecar_files=true
compress_xmp_tags=only large entries
omit_tag_hierarchy=false
plugins/lighttable/tagging/no_entry_completion=false
opencl=true
opencl_scheduling_profile=default
opencl_synch_cache=active module
opencl_library=
opencl_memory_requirement=768
opencl_size_roundup=16
maximum_number_tiles=10000
ask_before_remove=true
ask_before_delete=true
send_to_trash=true
ask_before_move=true
ask_before_copy=true
ask_before_rmdir=false
show_folder_levels=1
plugins/collection/film_id=1
plugins/collection/rating=1
plugins/collection/rating_comparator=3
plugins/collection/query_flags=3
plugins/collection/filter_flags=3
plugins/collection/descending=false
plugins/collection/sort=0
plugins/capture/jobcode=capture job
plugins/capture/storage/basedirectory=$(PICTURES_FOLDER)/darktable
plugins/capture/storage/subpath=$(YEAR)$(MONTH)$(DAY)_$(JOBCODE)
plugins/capture/storage/namepattern=$(YEAR)$(MONTH)$(DAY)_$(SEQUENCE).$(FILE_EXTENSION)
plugins/capture/camera/import/jobcode=noname
ui_last/colorpicker_mean=0
ui_last/fullscreen=false
ui_last/maximized=true
ui_last/grouping=false
ui_last/view=0
ui_last/window_x=0
ui_last/window_y=0
ui_last/window_w=900
ui_last/window_h=500
ui_last/panel_left=-1
ui_last/panel_right=-1
ui_last/panel_top=0
ui_last/panel_bottom=0
ui_last/expander_import=2
ui_last/import_ignore_jpegs=false
ui_last/import_recursive=false
ui_last/import_last_creator=
ui_last/import_last_publisher=
ui_last/import_last_rights=
ui_last/import_last_tags=
ui_last/import_last_directory=
ui_last/import_initial_rating=1
plugins/capture/mode=0
plugins/lighttable/filmstrip/visible=true
plugins/lighttable/filmstrip/height=120
plugins/lighttable/timeline/visible=true
plugins/lighttable/timeline/height=72
plugins/lighttable/timeline/last_zoom=0
plugins/lighttable/low_quality_thumbnails=false
plugins/lighttable/thumbnail_hq_min_level=720p
plugins/lighttable/extended_thumb_overlay=false
pressure_sensitivity=off
brush_smoothing=medium
channel_display=false color
plugins/darkroom/image_infos_pattern=$(EXIF_EXPOSURE) s • f/$(EXIF_APERTURE) • $(EXIF_FOCAL_LENGTH) mm • ISO $(EXIF_ISO)
plugins/darkroom/image_infos_position=bottom
plugins/darkroom/search_iop_by_text=show both
database_cache_quality=89
plugins/lighttable/draw_group_borders=true
session/base_directory_pattern=$(PICTURES_FOLDER)/Darktable
session/sub_directory_pattern=$(YEAR)$(MONTH)$(DAY)_$(JOBCODE)
session/filename_pattern=$(YEAR)$(MONTH)$(DAY)_$(SEQUENCE).$(FILE_EXTENSION)
plugins/lighttable/layout=1
plugins/lighttable/base_layout=1
plugins/lighttable/images_in_row=5
plugins/lighttable/culling_num_images=2
plugins/lighttable/culling_zoom_mode=0
plugins/lighttable/culling_last_id=-1
plugins/lighttable/recentcollect/num_items=0
plugins/lighttable/recentcollect/line0=
plugins/lighttable/recentcollect/line1=
plugins/lighttable/recentcollect/line2=
plugins/lighttable/recentcollect/line3=
plugins/lighttable/recentcollect/line4=
plugins/lighttable/recentcollect/line5=
plugins/lighttable/recentcollect/line6=
plugins/lighttable/recentcollect/line7=
plugins/lighttable/recentcollect/line8=
plugins/lighttable/recentcollect/line9=
plugins/lighttable/copy_history/pastemode=0
plugins/lighttable/collect/single-click=false
plugins/lighttable/collect/num_rules=0
plugins/lighttable/collect/item0=0
plugins/lighttable/collect/string0=
plugins/lighttable/collect/item1=0
plugins/lighttable/collect/string1=
plugins/lighttable/collect/item2=0
plugins/lighttable/collect/string2=
plugins/lighttable/collect/item3=0
plugins/lighttable/collect/string3=
plugins/lighttable/collect/item4=0
plugins/lighttable/collect/string4=
plugins/lighttable/collect/item5=0
plugins/lighttable/collect/string5=
plugins/lighttable/collect/item6=0
plugins/lighttable/collect/string6=
plugins/lighttable/collect/item7=0
plugins/lighttable/collect/string7=
plugins/lighttable/collect/item8=0
plugins/lighttable/collect/string8=
plugins/lighttable/collect/item9=0
plugins/lighttable/collect/string9=
plugins/lighttable/collect/mode0=0
plugins/lighttable/collect/mode1=0
plugins/lighttable/collect/mode2=0
plugins/lighttable/collect/mode3=0
plugins/lighttable/collect/mode4=0
plugins/lighttable/collect/mode5=0
plugins/lighttable/collect/mode6=0
plugins/lighttable/collect/mode7=0
plugins/lighttable/collect/mode8=0
plugins/lighttable/collect/mode9=0
plugins/darkroom/tagging/visible=true
plugins/lighttable/tagging/ask_before_delete_tag=true
plugins/lighttable/style/ask_before_delete_style=true
plugins/map/show_map_osd=true
plugins/map/show_outline=true
plugins/map/max_outline_nodes=10000
plugins/map/filter_images_drawn=FALSE
plugins/map/max_images_drawn=100
plugins/lighttable/metadata_view/pretty_location=true
plugins/lighttable/draw_custom_metadata=false
lighttable/ui/single_module=false
lighttable/ui/expose_statuses=false
darkroom/ui/single_module=true
darkroom/ui/activate_expand=false
lighttable/ui/scroll_to_module=false
lighttable/ui/preview/bottom_visible=false
lighttable/ui/preview/left_visible=false
lighttable/ui/preview/right_visible=false
lighttable/ui/preview/header_visible=false
lighttable/ui/preview/toolbar_bottom_visible=false
lighttable/ui/preview/toolbar_top_visible=false
darkroom/ui/scroll_to_module=true
darkroom/ui/sidebar_scroll_default=false
plugins/darkroom/ui/border_size=10
scrollbars=lighttable
panel_scrollbars_always_visible=true
ui_last/expander_metadata=0
ui_last/expander_navigation=-1
ui_last/expander_histogram=-1
ui_last/expander_snapshots=-1
ui_last/expander_history=-1
plugins/lighttable/export/force_lcms2=false
plugins/slideshow/high_quality=true
plugins/lighttable/export/high_quality_processing=false
rating_one_double_tap=false
darkroom/ui/rawoverexposed/mode=0
darkroom/ui/rawoverexposed/colorscheme=0
darkroom/ui/rawoverexposed/threshold=1.0
darkroom/ui/overexposed/colorscheme=1
darkroom/ui/overexposed/lower=2.0
darkroom/ui/overexposed/upper=98.0
plugins/darkroom/demosaic/fdc_xover_iso=1600
plugins/darkroom/denoiseprofile/show_compute_variance_mode=false
plugins/darkroom/demosaic/quality=at most PPG (reasonable)
plugins/lighttable/export/pixel_interpolator=lanczos3
plugins/lighttable/export/width=0
plugins/lighttable/export/height=0
plugins/lighttable/export/storage_name=disk
plugins/lighttable/export/format_name=jpeg
plugins/lighttable/export/style=
plugins/imageio/storage/disk/file_directory=$(FILE_FOLDER)/darktable_exported/$(FILE_NAME)
plugins/imageio/storage/gallery/file_directory=$(HOME)/darktable_gallery/img_$(SEQUENCE)
plugins/imageio/storage/gallery/title=darktable gallery
plugins/imageio/storage/email/client=
plugins/imageio/storage/gphoto/id=642055548087-n01fgvugnbns7a9jq8jfucjsn5l1t6so.apps.googleusercontent.com
plugins/imageio/storage/gphoto/secret=o29QcbsDWS5cauRqdmGdF3sP
plugins/imageio/format/pdf/size=a4
plugins/imageio/format/pdf/border=0 mm
plugins/imageio/format/pdf/dpi=300
plugins/imageio/format/pdf/compression=1
plugins/imageio/format/jpeg/quality=95
plugins/imageio/format/tiff/bpp=8
plugins/imageio/format/tiff/compress=2
plugins/imageio/format/tiff/compresslevel=5
plugins/imageio/format/png/bpp=8
plugins/imageio/format/png/compression=5
plugins/pwstorage/pwstorage_backend=auto
plugins/lighttable/export/icctype=-1
plugins/lighttable/export/iccprofile=
plugins/lighttable/export/iccintent=-1
plugins/darkroom/groups=2
plugins/darkroom/active=
plugins/darkroom/spots/circle_size=0.02
plugins/darkroom/spots/circle_border=0.02
plugins/darkroom/spots/ellipse_radius_a=0.02
plugins/darkroom/spots/ellipse_radius_b=0.01414
plugins/darkroom/spots/ellipse_border=0.02
plugins/darkroom/spots/ellipse_rotation=90.0
plugins/darkroom/spots/ellipse_flags=0
plugins/darkroom/spots/path_border=0.05
plugins/darkroom/spots/brush_border=0.05
plugins/darkroom/spots/brush_density=1.0
plugins/darkroom/spots/brush_hardness=0.66
plugins/darkroom/clipping/ratio_d=0
plugins/darkroom/clipping/ratio_n=0
plugins/darkroom/clipping/guide=0
plugins/darkroom/histogram/mode=logarithmic
plugins/darkroom/histogram/show_red=true
plugins/darkroom/histogram/show_green=true
plugins/darkroom/histogram/show_blue=true
plugins/darkroom/masks/opacity=1.0
plugins/darkroom/masks/circle/size=0.05
plugins/darkroom/masks/circle/border=0.05
plugins/darkroom/masks/ellipse/radius_a=0.05
plugins/darkroom/masks/ellipse/radius_b=0.03535
plugins/darkroom/masks/ellipse/border=0.05
plugins/darkroom/masks/ellipse/rotation=90.0
plugins/darkroom/masks/ellipse/flags=0
plugins/darkroom/masks/path/border=0.05
plugins/darkroom/masks/gradient/rotation=0.0
plugins/darkroom/masks/gradient/compression=0.5
plugins/darkroom/masks/gradient/steepness=0.0
plugins/darkroom/masks/brush/border=0.05
plugins/darkroom/masks/brush/density=1.0
plugins/darkroom/masks/brush/hardness=0.66
plugins/darkroom/ashift/near_delta=20.0
plugins/print/print/black_point_compensation=true
ui_last/display_profile_source=all
metadata/resolution=300
run_crawler_on_start=false
plugins/lighttable/audio_player=aplay
plugins/darkroom/lut3d/def_path=
plugins/darkroom/basecurve/auto_apply=true
plugins/darkroom/basecurve/auto_apply_percamera_presets=false
plugins/darkroom/sharpen/auto_apply=false
screen_dpi_overwrite=-1.0
screen_ppd_overwrite=-1.0
plugins/lighttable/preview/full_size_preload_count=1
plugins/lighttable/preview/max_in_memory_images=4
codepaths/sse2=true
codepaths/openmp_simd=false
allow_lab_output=false
plugins/darkroom/rawprepare/allow_editing_crop=false
plugins/map/geotagging_search_url=https://nominatim.openstreetmap.org/search/%s?format=xml&limit=%d&polygon_text=1
plugins/darkroom/temperature/visible=true
plugins/darkroom/exposure/visible=true
plugins/darkroom/colorbalance/visible=true
plugins/darkroom/colorin/visible=true
plugins/darkroom/clipping/visible=true
plugins/darkroom/sharpen/visible=true
plugins/darkroom/basecurve/visible=true
plugins/darkroom/highlights/visible=true
plugins/darkroom/demosaic/visible=true
plugins/darkroom/rgblevels/visible=true
plugins/darkroom/flip/visible=true
plugins/darkroom/bilat/visible=true
plugins/darkroom/denoiseprofile/visible=true
plugins/darkroom/hazeremoval/visible=true
plugins/darkroom/toneequal/visible=true
plugins/darkroom/lens/visible=true
darkroom/ui/scale_step_multiplier=1.0
darkroom/ui/scale_rough_step_multiplier=10.0
darkroom/ui/scale_precise_step_multiplier=0.1
performance_configuration_version_completed=0
```

**Comment by AlicVB** [comment]
Ok, I *think* I know where it comes from : TAB accel is completely buggy :(
I've normally fixed that in my last PR #3726 If you have some time, can you check that it works ?
Otherwise, when you have all panels closed, you can first press another time on TAB and then tweak them as you want. That should work...

**Comment by mtszsobczak** [comment]
Ahh thanks I didn't know about TAB shortcut! :) I will try to check your branch later today.
